bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -810,8 +828,10 @@ TableInfo createTableFromQuery() {
       if (job.getStatus().getError() != null) {
         throw BigQueryUtil.convertToBigQueryException(job.getStatus().getError());
       }
+      // Registering a cleanup job
+      CLEANUP_JOBS.add(() -> bigQueryClient.deleteTable(tempTable));
       // add expiration time to the table
-      TableInfo createdTable = bigQueryClient.getTable(destinationTable);
+      TableInfo createdTable = bigQueryClient.getTable(tempTable);
       long expirationTime =
           createdTable.getCreationTime() + TimeUnit.MINUTES.toMillis(expirationTimeInMinutes);
```

Comment:
Can we avoid this change by setting the expiration time during creation?

Reply:
Unfortunately no....